### PR TITLE
Include playbooks in main.yml

### PR DESCRIPTION
**Pull request by trishnaguha**
@gbraad Here are some improvements of the repo. We still have `rebase.yml` separate which I'm going to push later.

### CODE REVIEW

File: compose.yml [hunk 1]
```diff
@@ -1,58 +1,52 @@
...
+
+- debug: msg="{{ result.stdout }}"
+
+- name: Download daemon rpm
+  command:
+    wget http://libslack.org/daemon/download/daemon-0.6.4-1.x86_64.rpm
+  when: result|failed
+
+- name: Install daemon
+  command:
```
Comment (gbraad) [inline compose.yml]: packages can be installed with an ansible module...

File: compose.yml [hunk 2]
```diff
@@ -1,58 +1,52 @@
...
-  - name: Compose Fedora-Atomic Tree
-    command:
-      chdir=/workspace/fedora-atomic
-      rpm-ostree compose tree --cachedir=/srv/cache --repo=/srv/repo ./fedora-atomic-docker-host.json
-
-  - name: Running Http server at Port 35000, Use `ip addr` to check the IP Address...
-    shell:
-      /usr/local/bin/daemon -- ostree trivial-httpd -P 35000 --log-file - /srv
+- name: Install the list of required packages
+  package: name={{ item }} state=installed
```
Comment (gbraad) [inline compose.yml]: requirements in a separate playbook?

Comment (gbraad) [inline compose.yml]: requirements in a separate playbook?

Comment (trishnaguha) [reply compose.yml]: Sure.

File: compose.yml (hunk 1 above)
Comment (trishnaguha) [reply compose.yml]: @gbraad `daemon` module needs to downloaded first and then installed with rpm on fedora. It cannot be done with package or dnf module.

File: compose.yml [hunk 3]
```diff
@@ -1,58 +1,52 @@
...
+  - qemu-img
+
+- name: Query daemon package
+  shell:
+    rpm -q daemon
+  register: result
+
+- debug: msg="{{ result.stdout }}"
+
+- name: Download daemon rpm
```
Comment (gbraad) [inline compose.yml]: download to /tmp maybe ? Then you also do not need the ignore.

File: compose.yml [hunk 4]
```diff
@@ -1,58 +1,52 @@
...
+  command:
+    wget http://libslack.org/daemon/download/daemon-0.6.4-1.x86_64.rpm
+  when: result|failed
+
+- name: Install daemon
+  command:
+    rpm --install ./daemon-0.6.4-1.x86_64.rpm
```
Comment (gbraad) [inline compose.yml]: really not available in epel? or rpmfusion? hmmmm

**Review by gbraad — CHANGES_REQUESTED** [review]
Please see comments.

File: requirements.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+- name: Download daemon rpm
+  get_url:
+    url: http://libslack.org/daemon/download/daemon-0.6.4-1.x86_64.rpm
+    dest: /tmp
+
+- name: Install daemon
+  command:
+    rpm --install /tmp/daemon-0.6.4-1.x86_64.rpm
+  ignore_errors: yes
```
Comment (trishnaguha) [inline requirements.yml]: I'm using `ignore_errors` so that the playbook won't fail if `daemon` is already installed.

**Comment by trishnaguha** [comment]
@gbraad Pushed the changes.

File: requirements.yml (hunk 5 above)
Comment (gbraad) [reply requirements.yml]: you could consider combining all steps in one go...

```yaml
- name: Install daemon package
   command: >-
      wget ...
      rpm ....
      rm ...
   ignore_errors: yes
```

Comment (trishnaguha) [reply requirements.yml]: I thought to use `get_url` module instead of `wget command` ;-). Does it look good?

Comment (gbraad) [reply requirements.yml]: I considered proposing this, but sometimes it is a tradeoff to make. It is a single step that can be skipped with a tag in future... or do we want to be explicit and show each individual action. If you can make it work, please go ahead 👍

Comment (trishnaguha) [reply requirements.yml]: Cool. I'm fine with either option. Not a big deal. Let me update it.

**Comment by trishnaguha** [comment]
@gbraad How does this diff look on including `rebase.yml` ?

```
[tguha@localhost build-atomic-host]$ git diff
diff --git a/main.yml b/main.yml
index 8e676b7..f06608d 100644
--- a/main.yml
+++ b/main.yml
@@ -8,3 +8,13 @@
   - include: requirements.yml
diff --git a/main.yml b/main.yml
index 8e676b7..f06608d 100644
--- a/main.yml
+++ b/main.yml
@@ -8,3 +8,13 @@
   - include: requirements.yml
   - include: compose.yml
   - include: ssh-setup.yml
+
+- name: Rebase
+  hosts: atomichost
+  become: yes
+  become_method: sudo
+
+  tasks:
+  - include: rebase.yml
+    vars:
+      httpserver: 192.168.122.1
diff --git a/rebase.yml b/rebase.yml
index 02ff6b9..2da028d 100644
--- a/rebase.yml
+++ b/rebase.yml
@@ -1,14 +1,8 @@
 ---
-- name: Rebase On the Tree
-  hosts: atomichost
-  become: yes
-  become_method: sudo
+- name: Create remote named my-atomic
+  command:
+    ostree remote add my-atomic http://"{{ httpserver }}":35000/repo --no-gpg-verify
 
-  tasks:
-  - name: Create remote named my-atomic
-    command:
-      ostree remote add my-atomic http://'{{ httpserver }}':35000/repo --no-gpg-verify
-
-  - name: Rebase on to my tree
-    command:
-      rpm-ostree rebase my-atomic:fedora-atomic/25/x86_64/docker-host
+- name: Rebase on to my tree
+  command:
+    rpm-ostree rebase my-atomic:fedora-atomic/25/x86_64/docker-host
```

**Comment by trishnaguha** [comment]
Merging this in as now I have other changes to this repo which will not create `inventory` file anymore.
I'll create a separate PR for that. Thanks for the review :-).

**Pull request merged**